polaris-react/src/utilities/use-index-resource-state.ts

```diff
@@ -95,5 +95,16 @@ export function useIndexResourceState<T extends {[key: string]: unknown}>(
     [allResourcesSelected, resources, resourceIDResolver],
   );
 
+  useEffect(() => {
+    const resourceIds = resources.map(resourceIDResolver);
+    const selectedAvailableResources = selectedResources.filter((item) =>
+      resourceIds.includes(item),
+    );
```

Comment:
Using an array here for the ids means that we need to potentially scan the whole array for each `selectedResource`.

I'd expect `resources` to be relatively small, so this shouldn't make much difference in terms of performance, but I think using a `Set` here might be a better fit since we just want to check the existence of elements in the structure. 

I'd suggest something like:

```js
const resourceIds = new Set(resources.map(resourceIDResolver);
const selectedAvailableResources = selectedResources.filter((item) =>
  resourceIds.has(item),
);
```